src/services/PostService.jsx

```diff
@@ -16,6 +16,19 @@ export const getPosts = () => {
   });
 };
 
+export const getSinglePost = (postId) => {
+  return axios.get(`/posts/${postId}`).then((response) => {
+    if (response.status === 200) {
+      if (response.data && response.data.posts) {
+        return response.data;
```

Comment:
can we just do `return response.data.posts[0]` here so that the function that calls this is cleaner? And check that the length is greater than 0 just in case.